packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -700,15 +700,15 @@ export default function(
     instance.refs = emptyObject;
     instance.context = getMaskedContext(workInProgress, unmaskedContext);
 
-    warning(
-      instance.state !== instance.props,
-      '%s: It is not recommended to assign props directly to state' +
-        "because updates to props won't be reflected in state." +
-        'In most cases, it is better to use props directly.',
-      getComponentName(workInProgress) || 'A component',
-    );
-
     if (__DEV__) {
+      warning(
+        instance.state !== instance.props,
+        '%s: It is not recommended to assign props directly to state ' +
+          "because updates to props won't be reflected in state. " +
+          'In most cases, it is better to use props directly.',
+        getComponentName(workInProgress) || 'A component',
```

Comment:
nit: I think we generally use "Component" as the default name in warnings when it's followed by ":" (e.g. "Component: blah". We use "A component" when it's part of a sentence (e.g. "A component blah")